src/interceptors/response.ts

```diff
@@ -297,7 +303,11 @@ export function defaultResponseInterceptor(axios: AxiosCacheInstance): ResponseI
       }
 
       // Rejects all other requests waiting for this response
-      await rejectResponse(id, config);
+      await rejectResponse(
+        id,
+        config,
+        error.code !== 'ERR_CANCELED' || (error.code === 'ERR_CANCELED' && cache.state !== 'cached')
```

Comment:
Could you add a comment here saying why should we keep the cache of cancelled requests? I've read both your issue and PR and there's not an explanation for that :/ _(Having it as a comment in here will be great for future readers.)_

Reply:
I didn't see a reason not to. Quoting my issue: "Also, it seems harmless to keep the cache associated to a request, when it's cancelled, regardless of react strict mode. What do you think?"